src/plugins/base64/base64.c

```diff
@@ -0,0 +1,294 @@
+/**
+ * @file
+ *
+ * @brief filter plugin for the Base64 encoding
+ *
+ * @copyright BSD License (see doc/COPYING or http://www.libelektra.org)
+ *
+ */
+
+#ifndef HAVE_KDBCONFIG
+#include "kdbconfig.h"
+#endif
+#include "base64.h"
+#include <kdb.h>
+#include <kdberrors.h>
+#include <stdlib.h>
+#include <string.h>
+
+static const char * alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
+#define ALPHABET_LENGTH (64)
+static const char padding = '=';
+
+/**
+ * @brief read prefix from plugin configuration.
+ * @param config holds the plugin configuration.
+ * @returns the prefix to be used by the plugin.
+ */
+static const char * getPrefix (KeySet * config)
+{
+	Key * k = ksLookupByName (config, ELEKTRA_PLUGIN_BASE64_PARAM_PREFIX, 0);
+	if (k)
+	{
+		return keyString (k);
+	}
+	return ELEKTRA_PLUGIN_BASE64_DEFAULT_PREFIX;
+}
+
+/**
+ * @brief encodes arbitrary binary data using the Base64 encoding scheme (RFC4648)
+ * @param input holds the data to be encoded
+ * @param inputLength tells how many bytes the input buffer is holding.
+ * @returns an allocated string holding the Base64 encoded input data or NULL if the string can not be allocated. Must be freed by the caller.
+ */
+char * ELEKTRA_PLUGIN_FUNCTION (ELEKTRA_PLUGIN_NAME, base64Encode) (const kdb_octet_t * input, const size_t inputLength)
+{
+	size_t encodedLength;
+	if (inputLength % 3 == 0)
+	{
+		encodedLength = (inputLength / 3 * 4) + 1;
+	}
+	else
+	{
+		encodedLength = ((inputLength + (3 - (inputLength % 3))) / 3 * 4) + 1;
+	}
+
```

Comment:
how? `encodedLength` is given by the encoding schema. Do you mean something like `assert(encodedLength > 0)`?